packages/card/src/vaadin-card.js

```diff
@@ -46,7 +294,65 @@ class Card extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
 
   /** @protected */
   render() {
-    return html`<slot></slot>`;
+    return html`
+      <div part="media">
+        <slot name="media"></slot>
+      </div>
+      <div part="header">
+        <slot name="header-prefix"></slot>
+        <slot name="header">
+          <slot name="title"></slot>
+          <slot name="subtitle"></slot>
+        </slot>
+        <slot name="header-suffix"></slot>
+      </div>
+      <div part="content">
+        <slot></slot>
+      </div>
+      <div part="footer">
+        <slot name="footer"></slot>
+      </div>
+    `;
+  }
+
+  /** @private */
+  _mutationObserver;
+
+  /** @private */
+  _onMutation() {
+    // Chrome doesn't support `:host(:has())`, so we'll recreate that with custom attributes
+    this.toggleAttribute('has-media', this.querySelector(':scope > [slot="media"]'));
+    this.toggleAttribute('has-header', this.querySelector(':scope > [slot="header"]'));
+    this.toggleAttribute(
+      'has-title',
+      this.querySelector(':scope > [slot="title"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute(
+      'has-subtitle',
+      this.querySelector(':scope > [slot="subtitle"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute('has-header-prefix', this.querySelector(':scope > [slot="header-prefix"]'));
+    this.toggleAttribute('has-header-suffix', this.querySelector(':scope > [slot="header-suffix"]'));
+    this.toggleAttribute('has-content', this.querySelector(':scope > :not([slot])'));
+    this.toggleAttribute('has-footer', this.querySelector(':scope > [slot="footer"]'));
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (!CSS.supports('selector(:host(:has([slot])))')) {
+      if (!this._mutationObserver) {
+        this._mutationObserver = new MutationObserver(this._onMutation.bind(this));
+      }
+      this._mutationObserver.observe(this, { childList: 'true', attributeFilter: ['slot'], subtree: true });
+      // Trigger once manually to update attributes on first render
+      this._onMutation();
```

Comment:
I suppose we could also use the `slotchange` event. But does it bubble, if we could manage with just one event listener?

Reply:
I’m actually going to replace the MutationObserver with a slotchange listener.